math_notes/mathnotes.py

```diff
@@ -0,0 +1,208 @@
+        path = Path("./math_notes/predictions/")
+        filepath = path / filename
+
+        try:
+            with open(filepath, newline="") as csvfile:
+                csv_reader = csv.reader(csvfile)
+                csv_reader.__next__()
+                for row in csv_reader:
+                    latex_readin.append(row[0])
+        finally:
+            return latex_readin
+
+    def _save_posn(event):
+        """Saves positional coordinates while drawing on canvas.
+
+        :param event: Recorded mouse-click events for drawing.
+        :type event: class tkinter.Event
+
+        :return: none
+        :rtype: none
+        """
+
+        global lastx, lasty
+        lastx, lasty = event.x, event.y
+
+    def _add_line(event):
+        """Adds a line connection previous location while drawing to current location.
+
+        Event here is where the mouse was and is located. Also draws the same line
+        on a PIL image which represents the image drawn on the tkinter canvas by the
+        user. This is the image that will actually be saved and used by the OCR.
+
+        :param event: Recorded mouse-click events for drawing.
+        :type event: class tkinter.Event
+
+        :return: none
+        :rtype: none
+        """
+
+        # This canvas call is what the user sees on the screen.
+        canvas.create_line(
+            (lastx, lasty, event.x, event.y),
+            smooth=True,
+            width=linewidth,
+            fill=linecolor,
+        )
+
+        # The draw call is in the background (invisible)
+        # capturing what will actually get converted to an image.
+        draw.line(
+            [lastx, lasty, event.x, event.y],
+            fill=linecolor,
+            width=linewidth,
+            joint="curve",
+        )
+        _save_posn(event)
+
+    global lastx, lasty
+
+    root = Tk()
+    root.title("Math Canvas")
+
+    text = tk.StringVar()
+    text.set("Prediction (ready): ")
+
+    # Define the supporting paths.
+    filename["path"] = filename.get("path", Path("./math_notes/temp_files/"))
+
+    # Create the canvas.
+    canvas_dimensions = str(width) + "x" + str(int(height * 1.5))
+    root.geometry(canvas_dimensions)
+
+    # Instantiate the tkinter canvas for users to draw on.
+    canvas = Canvas(root, bg="white", width=width, height=height)
+    canvas.grid(row=0, column=0)
+
+    # PIL create an empty image and draw object to memory only.
+    # It is not visible.
+    canvas_image = Image.new("RGB", (width, int(height * 1.5)), (255, 255, 255))
+    draw = ImageDraw.Draw(canvas_image)
+
+    # Capturing mouse motion.
+    canvas.bind("<Button-1>", _save_posn)
+    canvas.bind("<B1-Motion>", _add_line)
+
+    # Labels for readouts.
+    row = 10
+    label = tk.Label(root, textvariable=text, font=("helvetica", 18))
+    label.grid(row=row + 3, column=0)
+
+    # Buttons to save canvas, quit canvas, browse images.
+    myFont = font.Font(family="SFMono-Regular", size=11)
```

Comment:
Not PEP-8 style. You probably stole this from the internet. Run through this code and make sure everything follows best practices, including PEP-8, good variable names, object named is actually being used in the code, etc.

Reply:
Rearchitected this to PEP-8. I assume this was about the way I used label.grid itself, but maybe you meant the naming too? The name is now changed to reflect this anyhow.